### PR TITLE
Prevent race conditions in concurrent MLS commit requests

### PR DESCRIPTION
**Pull request by stefanwire**
https://wearezeta.atlassian.net/browse/FS-437

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [x] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.

### CODE REVIEW

File: services/galley/test/integration/API/MLS/Util.hs [hunk 1]
```diff
@@ -353,6 +353,7 @@ claimKeyPackage brig claimant target =
         . zUser claimant
     )
 
+-- TODO(SB) generalise such that two prepared commits can be sent
```
Comment (smatting) [inline services/galley/test/integration/API/MLS/Util.hs:356]: TODO left

**Review by pcapriotti — APPROVED** [review]
Looks good if CI agrees. Minor comments follow.

File: services/galley/src/Galley/Cassandra/Queries.hs [hunk 2]
```diff
@@ -370,6 +370,12 @@ addLocalMLSClients = "update member set mls_clients = mls_clients + ? where conv
 addRemoteMLSClients :: PrepQuery W (C.Set ClientId, ConvId, Domain, UserId) ()
 addRemoteMLSClients = "update member_remote_user set mls_clients = mls_clients + ? where conv = ? and user_remote_domain = ? and user_remote_id = ?"
 
+acquireCommitLock :: PrepQuery W (GroupId, Epoch, Int32) Row
+acquireCommitLock = "insert into mls_commit_locks (group_id, epoch) values (?, ?) if not exists using ttl ?"
```
[the remark this answers is not in the document]
Comment (stefanwire) [reply services/galley/src/Galley/Cassandra/Queries.hs:374]: We tried it locally by temporarily removing the finishing part of the bracket and observed the entries appearing and finally disappearing after the TTL expired. Would this manual test count or should I look deeper into it?

**Comment by smatting** [comment]
Not sure if this improves things. If we add a timeout and the commit processing can't meet it, then it's quite likely we're left with a broken state. If we don't add a timeout and the commit processing doesn't finish in time we're losing the guarantee the lock gives, but then there's still the chance that no commit arrives before the commit processing is finished.

Alternatively we could increase the TTL of the lock to make this problem less likely to occur.

**Review by smatting — APPROVED** [review]

**Pull request merged**